iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -49,8 +48,7 @@ internal class AmqpIotTransport : IDisposable
             {
                 TargetHost = hostName,
                 Certificate = null,
-                CertificateValidationCallback = _amqpTransportSettings.RemoteCertificateValidationCallback
-                    ?? OnRemoteCertificateValidation,
+                CertificateValidationCallback = OnRemoteCertificateValidation,
```

Comment:
Ah, this looks like a gap in our plan. The remote certificate validation callback is being set into the TLS settings, so we'll need a way for users to be able to pass this to us for tcp-based connections as well.